src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/BasicSpringDataRepositoryDatabaseService.java

```diff
@@ -32,64 +31,54 @@
- * {@link AccessioningRepository}, a function to generate the entities from the triple model/hash/accession, a function
+ * {@link IAccessionedObjectRepository}, a function to generate the entities from the triple model/hash/accession, a function
  * to get the accession from the entity and a function to get the hashed representation of the message from the entity.
  *
  * @param <MODEL>
  * @param <ENTITY>
- * @param <HASH>
  * @param <ACCESSION>
  */
-public class BasicSpringDataRepositoryDatabaseService<MODEL, ENTITY extends MODEL, HASH, ACCESSION extends Serializable>
-        implements DatabaseService<MODEL, HASH, ACCESSION> {
+public class BasicSpringDataRepositoryDatabaseService<MODEL, ENTITY extends IAccessionedObject<ACCESSION>,
+        ACCESSION extends Serializable> implements DatabaseService<MODEL, String, ACCESSION> {
 
-    private AccessioningRepository<ENTITY, HASH, ACCESSION> repository;
+    private IAccessionedObjectRepository<ENTITY, String, ACCESSION> repository;
 
-    private final Function<ModelHashAccession<MODEL, HASH, ACCESSION>, ENTITY> toEntityFunction;
+    private final Function<ModelHashAccession<MODEL, String, ACCESSION>, ENTITY> toEntityFunction;
 
-    private final Function<ENTITY, ACCESSION> getAccessionFunction;
+    private final Function<ENTITY, MODEL> toModelFunction;
 
-    private final Function<ENTITY, HASH> getHashedMessageFunction;
-
-
-    public BasicSpringDataRepositoryDatabaseService(AccessioningRepository<ENTITY, HASH, ACCESSION> repository,
-                                                    Function<ModelHashAccession<MODEL, HASH, ACCESSION>, ENTITY> toEntityFunction,
-                                                    Function<ENTITY, ACCESSION> getAccessionFunction,
-                                                    Function<ENTITY, HASH> getHashedMessageFunction) {
+    public BasicSpringDataRepositoryDatabaseService(IAccessionedObjectRepository<ENTITY, String, ACCESSION> repository,
+                                                    Function<ModelHashAccession<MODEL, String, ACCESSION>, ENTITY> toEntityFunction,
+                                                    Function<ENTITY, MODEL> toModelFunction) {
         this.repository = repository;
         this.toEntityFunction = toEntityFunction;
-        this.getAccessionFunction = getAccessionFunction;
-        this.getHashedMessageFunction = getHashedMessageFunction;
+        this.toModelFunction = toModelFunction;
     }
 
     @Override
-    public Map<ACCESSION, MODEL> findAllAccessionsByHash(Collection<HASH> hashes) {
+    public Map<ACCESSION, MODEL> findAllAccessionsByHash(Collection<String> hashes) {
         return repository.findByHashedMessageIn(hashes).stream()
-                .collect(Collectors.toMap(getAccessionFunction, e -> e));
+                .collect(Collectors.toMap(IAccessionedObject::getAccession, toModelFunction));
```

Comment:
will this work with our legacy table where accessions are not unique? I mean, we won't generate them, but the DB will exist and we must be able to query it. Same applies to `Map<ACCESSION, MODEL> findAllAccessionMappingsByAccessions(List<ACCESSION> accessions)`

Reply:
it's not a problem of this PR, so I'm approving